src/cdk/tree/tree.md

```diff
@@ -0,0 +1,157 @@
+The `<cdk-tree>` enable developers to build customized tree for structured data.  The cdk-tree
+provides a foundation to build other features such as filtering on top of tree.
+For a Material design styled tree, see `<mat-tree>` which builds on top of the `<cdk-tree>`.
+
+There are two types of trees: Flat tree and Nested Tree. The DOM structures are different for
+these two types of trees.
+
+#### Flat tree
+
+In a flat tree, the hierarchy is flattened; nodes are not rendered inside of each other, but instead
```

Comment:
There's some duplication between the cdk docs and the mat docs. Would it make sense to distinguish these similar to the table docs?